### ISSUE TITLE
Research parameters:

### ISSUE THREAD

**Issue by GreNait** [GreNait/technical-thesis#19]
- [ ] What different activation functions are available and usable for the model
- [x] What loss functions/optimizers are available
- [x] What amount of layers with how many neurons make sense? <- Practical test
- [ ] Comparison between those different models
- [ ] addin Bounding box polygons to the detected figure - Labelling data https://github.com/tzutalin/labelImg
- [ ] best practice with ML
- [ ] logits of a model (model prediction) https://developers.google.com/machine-learning/glossary#logits
- [ ]  what is SparseCategoricalCrossentropy and what are others? https://www.tensorflow.org/tutorials/keras/classification
- [ ] what is leaky relu ?  https://www.tensorflow.org/api_docs/python/tf/nn/leaky_relu?hl=lt

**Comment by GreNait** [comment]
Flatten() needs to be done before have fully conected Layers but after a convolution. Flatten() makes a threediemsnional array flat for a vecctorisation. A model does not need the 3dimensional structure, convolutions do. See:
https://stackoverflow.com/questions/44572141/why-do-we-flatten-the-data-before-we-feed-it-into-tensorflow

**Comment by GreNait** [comment]
A guide to activation functions https://missinglink.ai/guides/neural-network-concepts/7-types-neural-network-activation-functions-right/

**Comment by GreNait** [comment]
It appears, that it is very important to not have an activation function at the last layer. I used for the small model test an softmax at the end. The accuracy was than stuck to 33%.
